internal/key/key.go

```diff
@@ -68,63 +53,46 @@ func NodeSSHDConfigEncoded() string {
 	return base64.StdEncoding.EncodeToString([]byte(nodeSSHDConfig))
 }
 
-func CAPAClusterOwnedTag(clusterName string) string {
-	return fmt.Sprintf("tag:sigs.k8s.io/cluster-api-provider-aws/cluster/%s", clusterName)
-}
-
-func FlatcarAWSAccountID(awsRegion string) string {
-	if strings.Contains(awsRegion, "cn-") {
-		return FlatcarChinaAMIOwner
-	} else {
-		return FlatcarAMIOwner
+func ValidateName(name string, enableLongNames bool) (bool, error) {
+	maxLength := NameLengthShort
+	if enableLongNames {
+		maxLength = NameLengthLong
 	}
-}
 
-func GenerateID() string {
-	compiledRegexp, _ := regexp.Compile("^[a-z]+$")
+	pattern := fmt.Sprintf("^[a-z][a-z0-9]{0,%d}$", maxLength-1)
+	matched, err := regexp.MatchString(pattern, name)
+	return matched, microerror.Mask(err)
+}
 
-	/* #nosec G404 */
+func GenerateName(enableLongNames bool) (string, error) {
 	for {
-		letterRunes := []rune(IDChars)
-		b := make([]rune, IDLength)
+		letterRunes := []rune(NameChars)
+		length := NameLengthShort
+		if enableLongNames {
+			length = NameLengthLong
+		}
+		characters := make([]rune, length)
 		rand.Seed(time.Now().UnixNano())
-		for i := range b {
-			b[i] = letterRunes[rand.Intn(len(letterRunes))]
+		for i := range characters {
+			characters[i] = letterRunes[rand.Intn(len(letterRunes))] //nolint:gosec
 		}
 
-		id := string(b)
+		generatedName := string(characters)
 
-		if _, err := strconv.Atoi(id); err == nil {
-			// ID is made up of numbers only, which we want to avoid.
+		if valid, err := ValidateName(generatedName, enableLongNames); err != nil {
```

Comment:
Reusing the same validation function as we use to validate a user-provided name instead of having a separate regex.